Engine/Code/Engine/Graphics/Vertex.h

```diff
@@ -0,0 +1,65 @@
+﻿#pragma once
+
+#include "Engine/Core/CoreIncludes.h"
+#include "Engine/Graphics/GLFunctions.h"
+
+enum DATATYPE {
+	APEX_FLOAT           = GL_FLOAT,
+	APEX_UNSIGNED_INT    = GL_UNSIGNED_INT,
+	APEX_UNSIGNED_BYTE   = GL_UNSIGNED_BYTE
+};
+
+struct VertexBufferElement
+{
+	unsigned int m_Type;
+	unsigned int m_ElementCount;
+	unsigned char m_Normalized;
+
+	static unsigned int GetSizeOfType(unsigned int type)
+	{
+		switch (type)
+		{
+		case APEX_FLOAT:           return 4;
+		case APEX_UNSIGNED_INT:    return 4;
+		case APEX_UNSIGNED_BYTE:   return 1;
+		}
+
+		return 0;
+	}
+
+	VertexBufferElement(unsigned int type, unsigned int count, unsigned char normalized);
+};
+
+struct VertexBufferLayout
+{
+	std::vector<VertexBufferElement> m_Element;
+	unsigned int m_Stride;
+
+	VertexBufferLayout(const std::vector<VertexBufferElement> element, unsigned int stride);
+};
+
+struct VertexMaster
+{
+	Vec3 m_Pos;
+	Vec4 m_Color;
+	Vec2 m_UV;
+
+	VertexMaster() {}
+	VertexMaster(const Vec3& position, const Vec4& color, const Vec2& uv) : m_Pos(position), m_Color(color), m_UV(uv) {}
+	~VertexMaster() {}
+};
+
+struct VertexPCU
+{
+	Vec3 m_Pos;
+	Vec4 m_Color;
```

Comment:
@kevin30061994 Color aas floats and not uints ??

Reply:
Well thats fine as long as the vertex buffer elments are created as such

Reply:
I don't think he has a color struct yet to do the uint split